tests/failpoint/db_failpoint_test.go

```diff
@@ -155,3 +155,115 @@ func TestFailpoint_LackOfDiskSpace(t *testing.T) {
 	require.Error(t, err)
 	require.ErrorIs(t, err, errors.ErrTxClosed)
 }
+
+// TestIssue72 reproduces issue 72.
+//
+// When application inserts key/value, that key can be changed by application
+// during insertion. bbolt should copy key before seeking and updating. Otherwise,
+// some branch node could point to freed page.
```

Comment:
It doesn't necessarily cause elements to point freed pages. In this test case, if you change the key to a larger key e.g. 200, the case will pass.

```suggestion
// When bbolt is processing a `Put` invocation, the key might be concurrently
// updated by the application which calls the `Put` API (although it shouldn't).
// It might lead to a situation that bbolt use an old key to find a proper
// position to insert the key/value pair, but actually inserts a new key.
// Eventually it might break the rule that all keys should be sorted. In a
// worse case, it might cause page elements to point to already freed pages.
```

Reply:
Thanks for the comment. Updated